kotlin/generate-offline-map/src/main/java/com/esri/arcgisruntime/sample/generateofflinemap/MainActivity.kt

```diff
@@ -0,0 +1,233 @@
+    takeMapOfflineButton.isEnabled = false
+
+    // create a portal item with the itemId of the web map
+    val portal = Portal(getString(R.string.portal_url), false)
+    val portalItem = PortalItem(portal, getString(R.string.item_id))
+    // create a map with the portal item
+    val map = ArcGISMap(portalItem)
+
+    map.addDoneLoadingListener {
+      if (map.loadStatus == LoadStatus.LOADED) {
+        // limit the map scale to the largest layer scale
+        map.apply {
+          maxScale = operationalLayers[6].maxScale
+          minScale = operationalLayers[6].minScale
+        }
+      } else {
+        val error = "Map failed to load: " + map.loadError.message
+        Toast.makeText(this, error, Toast.LENGTH_LONG).show()
+        Log.e(TAG, error)
+      }
+    }
+
+    // set the map to the map view
+    mapView.map = map
+
+    // update the download area box whenever the viewpoint changes
+    mapView.addViewpointChangedListener {
+      if (map.loadStatus == LoadStatus.LOADED) {
+        // enable the map offline button only after the map is loaded
+        if (!takeMapOfflineButton.isEnabled) takeMapOfflineButton.isEnabled = true
+        // upper left corner of the area to take offline
+        val minScreenPoint = Point(200, 200)
+        // lower right corner of the downloaded area
+        val maxScreenPoint = Point(
+          mapView.width - 200,
+          mapView.height - 200
+        )
+        // convert screen points to map points
+        val minPoint = mapView.screenToLocation(minScreenPoint)
+        val maxPoint = mapView.screenToLocation(maxScreenPoint)
+        // use the points to define and return an envelope
+        if (minPoint != null && maxPoint != null) {
+          val envelope = Envelope(minPoint, maxPoint)
+          downloadArea.geometry = envelope
+        }
+      }
+    }
+
+    // when the button is clicked, start the offline map task job
+    takeMapOfflineButton.setOnClickListener { generateOfflineMap() }
+  }
+
+  /**
+   * Use the generate offline map job to generate an offline map.
+   */
+  private fun generateOfflineMap() {
+    // delete any offline map already in the cache
+    val tempDirectoryPath: String =
+      cacheDir.toString() + File.separator.toString() + "offlineMap"
+    File(tempDirectoryPath).deleteRecursively()
+
+    // specify the extent, min scale, and max scale as parameters
+    var minScale: Double = mapView.mapScale
+    val maxScale: Double = mapView.map.maxScale
+    // minScale must always be larger than maxScale
+    if (minScale <= maxScale) {
+      minScale = maxScale + 1
+    }
+
+    val generateOfflineMapParameters = GenerateOfflineMapParameters(
+      downloadArea.geometry, minScale, maxScale
+    )
+    // create an offline map offlineMapTask with the map
+    val offlineMapTask = OfflineMapTask(mapView.map)
+
+    // create an offline map job with the download directory path and parameters and start the job
+    val job: GenerateOfflineMapJob =
```

Comment:
shouldn't need to declare the class here with Kotlin: you could rename the variable to be offlineMapJob if it's there for readability? 
```suggestion
    val job =
```

Reply:
that sounds reasonable to me!